### PR TITLE
An extension to view Angle's D3D11 resource map

### PR DESCRIPTION
**Pull request by sushraja-msft**
This extension dumps the resource table within angle, this is useful to translate gl ids into actual directx textures. This extension only works on windows and when Angle uses a D3D11 backend.

### CODE REVIEW

File: extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+"use strict";
+
+var AngleTextures = undefined;
+Loader.OnLoad(function () {
+  AngleTextures = {
+    GetRoots: function () {
+      return DbgObject.global("libGLESv2", "gSingleThreadedContext").catch((error)=>{
+        var errorMessage = ErrorMessages.CreateErrorsList(error) +
+                    ErrorMessages.CreateErrorReasonsList("Extension works in windows only and when Angle uses D3D11 backend.", ErrorMessages.SymbolsUnavailable);
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js]: Going forward, we should find a more declarative way of handling platform-specific extensions. Ideally, this extension wouldn't even show up on non-Windows platform. Perhaps we need to add something in extension.json. Can you file an issue for this?

File: extensions/target-specific/chromium/win-d3d11-angletextures/extension.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "win-d3d11-angletextures",
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/extension.json]: nit: please use the same naming format as the other extensions (WinD3D11AngleTextures)

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Inspect Compositor Frame</title>
+    <script type="text/javascript" src="/jsdbg/loader.js"></script>
+    <script type="text/javascript">
+        Loader.OnPageReady(() => {
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: You should call Chromium.SetTargetProcess(Chromium.GpuProcessName) here.

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                AngleTextures.ShowTextures(context) 
+            }, 
+            function (error) {
+                var rootsError = document.getElementById("rootsError");
+                rootsError.className = "error show";
+                rootsError.innerHTML = error;
+            });
+        });
+    </script>
+    <style>
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: Consider moving these styles into a .css file.

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    MIT License
+
+    Copyright (c) Microsoft Corporation. All rights reserved.
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Inspect Compositor Frame</title>
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: nit: consider using the extension name (or something similar) for the title

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    <script type="text/javascript" src="/jsdbg/loader.js"></script>
+    <script type="text/javascript">
+        Loader.OnPageReady(() => {            
+            AngleTextures.GetRoots().then((context) => {   
+                document.getElementById("tables").className = "show";              
+                AngleTextures.ShowTextures(context) 
+            }, 
+            function (error) {
```
Comment (sanketj) [inline extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: nit: "(error) =>", to be consistent

File: extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js (hunk 1 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js]: I dont think this is tenable, for this extension you would need to know if Chrome is using SwiftShader or D3D11backend which you cannot tell unless you break in on the other hand if this is a general concept then If JsDBG intends to be untied from Chromium/Edge I don't think marking extensions as per platform would be important.

File: extensions/target-specific/chromium/win-d3d11-angletextures/extension.json (hunk 2 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/extension.json]: I think I am consistent with gfx-type-helpers, ax-tree, chromium-helpers. In cases where the words are merged they refer to class names in code ?

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html (hunk 3 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: done

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html (hunk 4 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: done

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html (hunk 5 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: done

File: extensions/target-specific/chromium/win-d3d11-angletextures/index.html (hunk 6 above)
Comment (sushraja-msft) [reply extensions/target-specific/chromium/win-d3d11-angletextures/index.html]: done

File: extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js (hunk 1 above)
Comment (sanketj) [reply extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js]: Ideally, Windows-specific extensions shouldn't even be loaded/served on Mac/Linux (and vice versa). It's not just this one, we also have TEB and may have more in the future. They just add clutter on other platforms, so it's best to hide them from the user.

Comment (sanketj) [reply extensions/target-specific/chromium/win-d3d11-angletextures/AngleTextures.js]: We do something similar to hide the mshtml/edgehtml extensions. Seeing those in your view while debugging Chromium would be annoying. :)

File: extensions/target-specific/chromium/win-d3d11-angletextures/extension.json (hunk 2 above)
Comment (sanketj) [reply extensions/target-specific/chromium/win-d3d11-angletextures/extension.json]: Both "gfx-type-helpers" and "chromium-helpers" are headless extensions (won't be directly visible to the user), so those are fine. For AXTree, though the file is called "ax-tree", the extension name is "AXTree" and that's what's shown on the JsDbg landing page. Please follow the same style.

**Pull request merged**